moshi/src/test/java/com/squareup/moshi/MoshiTest.java

```diff
@@ -533,6 +533,26 @@
     assertThat(adapter.toJson(null)).isEqualTo("null");
   }
 
+  @Test public void upperBoundedWildcardsAreNotHandled() {
+    Moshi moshi = new Moshi.Builder().build();
+    try {
+      moshi.adapter(Types.subtypeOf(String.class));
```

Comment:
There could be a delegating factory that gives the string JsonAdapter for this type, but I don't see a use case.

Would somebody with a `List<?> objects` expect the object JsonAdapter to handle this? I doubt it, and I haven't seen anybody ask for it.